benchmarking/runs/mutations/mutation_benchmarking.py

```diff
@@ -0,0 +1,208 @@
+            "mutate_random_slice": [RandomSliceMutation(max_spread)],
+            "mutate_half_random_slice": [
+                RandomSliceMutation(num_mutations=(max_spread // 2)),
+                RandomSwapMutation(num_mutations=(max_spread // 2)),
+            ],
+            "mutate_greedy_local_max_n_2": [
+                GreedyLocalMaxMutation(number_of_teams=2, num_mutations=max_spread)
+            ],
+            "mutate_greedy_local_max_n_4": [
+                GreedyLocalMaxMutation(number_of_teams=4, num_mutations=max_spread)
+            ],
+            "mutate_greedy_local_max_n_8": [
+                GreedyLocalMaxMutation(number_of_teams=8, num_mutations=max_spread)
+            ],
+            "mutate_greedy_local_max_with_random_swap": [
+                GreedyLocalMaxMutation(num_mutations=(max_spread // 2)),
+                RandomSwapMutation(num_mutations=(max_spread // 2)),
+            ],
+            "mutate_greedy_local_max_with_random_slice_n_2": [
+                GreedyLocalMaxMutation(num_mutations=(max_spread // 2)),
+                RandomSliceMutation(num_mutations=(max_spread // 2)),
+            ],
+            "mutate_greedy_local_max_with_random_slice_n_4": [
+                GreedyLocalMaxMutation(
+                    number_of_teams=4, num_mutations=(max_spread // 2)
+                ),
+                RandomSliceMutation(max_spread // 2),
+            ],
+            "mutate_greedy_local_max_with_random_slice_n_8": [
+                GreedyLocalMaxMutation(
+                    number_of_teams=8, num_mutations=(max_spread // 2)
+                ),
+                (RandomSliceMutation(), max_spread // 2),
+            ],
+            "mutate_robinhood": [RobinhoodMutation(max_spread)],
+            "mutate_robinhood_half_random_swap": [
+                RobinhoodMutation(max_spread // 2),
+                RandomSwapMutation(max_spread // 2),
+            ],
+            "mutate_robinhood_holistic": [RobinhoodHolisticMutation(max_spread)],
+            "mutate_robinhood_holistic_half_random_swap": [
+                RobinhoodHolisticMutation(max_spread // 2),
+                RandomSwapMutation(max_spread // 2),
+            ],
+        }
+
+        artifacts: Dict[int, Dict[str, SimulationSetArtifact]] = {}
+
+        for class_size in class_sizes:
+            if class_size not in artifacts:
+                artifacts[class_size] = {}
+            for mutation_name, mutations in mutation_sets.items():
+                artifacts[class_size][mutation_name] = SimulationSet(
+                    settings=SimulationSettings(
+                        scenario=scenario,
+                        student_provider=RealisticMockStudentProvider(class_size),
+                        cache_key=f"mutations/mutation_benchmarking/{mutation_name}/class_size_{class_size}/",
+                        initial_teams_provider=RealisticMockInitialTeamsProvider(
+                            class_size // team_size
```

Comment:
another thought on floor division - not sure if this makes sense when the numbers don't divide nicely, no need to change anything but take a double glance at it